fix: Update Dockerfile (#2962)
ansys/pyfluent: doc/source/user_guide/make_container_image.rst

.. _ref_make_container_image:

Containerization of Fluent
==========================

.. warning:: You need a valid Ansys license to follow the steps in this section.

This document provides instructions and guidelines on how to containerize
Fluent for efficient and secure deployment and management.

Prerequisites
-------------

* A Linux machine with `Docker <https://www.docker.com>`_ installed.

* A valid Ansys license. Your Ansys reseller should have provided you with one.

* The following provided files:
  
  * `Dockerfile <https://github.com/ansys/pyfluent/blob/main/docker/fluent/Dockerfile>`_
  * `copy_docker_files.py <https://github.com/ansys/pyfluent/blob/main/docker/fluent/copy_docker_files.py>`_

Procedure
---------

> These text files indicate the files that are included or excluded during the copying for Ansys version 24R1.

Reviewer: These text files? Which?

Author: Removed. Thank you.

If you have a different version of Ansys then update these paths in the following text files accordingly, for example if you have

> the pre-installed Ansys version 23R2 then replace v241 with v232 and 24.1.0 with 23.2.0 etc.

Reviewer: _etc_ at the end is too vague to be included in instructions. take this list:
```
v241/CEI/BUILDINFO.txt
v241/CEI/apex241
v241/CEI/bin
v241/CEI/ensight241lang
v241/CEI/nexus241
```
Those embedded `241`s must also be updated, but are tey covered by the etc or not? - as a user, I would have a doubt.

Reviewer 2: Maybe, we can publish separate filelists for each release. Files can be added/deleted in newer version, e.g.

Author: Removed. Thank you.

1. Set current working directory
++++++++++++++++++++++++++++++++

* If you have cloned `PyFluent <https://github.com/ansys/pyfluent>`_ locally then change the current working directory to any of the
`fluent_222 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_222>`_, `fluent_231 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_231>`_,
`fluent_232 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_232>`_, `fluent_241 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_241>`_,
`fluent_242 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_242>`_ directories according to the Ansys version you have.

* If you haven't cloned `PyFluent <https://github.com/ansys/pyfluent>`_ locally then change the current working directory to the folder into which you have downloaded the files from any of the `fluent_222 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_222>`_, `fluent_231 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_231>`_,
`fluent_232 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_232>`_, `fluent_241 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_241>`_,

> `fluent_242 <https://github.com/ansys/pyfluent/blob/main/docker/fluent_242>`_ directories according to the Ansys version you have.

Author: Removed. Thank you.

.. _ref_make_container:

> Create Fluent Docker container image
> ====================================

Reviewer: ```suggestion
Containerization of Fluent
==========================
```

Author: Done. Thank you.

.. warning:: You need a valid Ansys license and an Ansys account to
   follow the steps in this section.

The following procedure for creating a Fluent Docker container image uses a

> Linux machine to copy the needed files from the Fluent installation directory.

Reviewer: license and linux machine are a subset of the (imminent) prerequisites. Omit them here and keep it to the point: outline the purpose of the document.

"This document provides instructions and guidelines on how to containerize Fluent for efficient and secure deployment and management."

Author: Done. Thank you.

Prerequisites
-------------

* A Linux machine with `Docker <https://www.docker.com>`_ installed.

> * A valid Ansys license and account. Your Ansys reseller should have provided you with one.

Reviewer: "with one" refers to what? There are two things mentioned in the previous sentence.

Author: We need license only as the Ansys is pre-installed. Done. Thank you.

* The following provided files:
  
  * `Dockerfile <https://github.com/ansys/pyfluent/blob/main/docker/fluent/Dockerfile>`_
  * `copy_docker_files.py <https://github.com/ansys/pyfluent/blob/main/docker/fluent/copy_docker_files.py>`_

Procedure
---------

* If you have cloned `PyFluent <https://github.com/ansys/pyfluent>`_ locally then change the current working directory to
`Docker files <https://github.com/ansys/pyfluent/blob/main/docker/fluent>`_ before executing these commands.

* If you haven't cloned `PyFluent <https://github.com/ansys/pyfluent>`_ locally then copy `Docker files <https://github.com/ansys/pyfluent/blob/main/docker/fluent>`_ into an empty folder and
execute these commands from that folder.

Copy needed files
+++++++++++++++++

Specify the pre-installed Ansys directory as a command line argument and run this script to copy needed files from the

> Ansys installation directory to the container image:

Reviewer: Re "to the container image" aren't they copied to the current directory? The container image doesn't exist at this point.

Author: Yes. Done. Thank you.

.. code:: python

    python copy_docker_files.py <path to 'ansys_inc' directory>

Not all the installation files are copied. These files indicate the files that are ignored during the copying:

  * `excludeCEIList.txt <https://github.com/ansys/pyfluent/blob/main/docker/fluent/excludeCEIList.txt>`_
  * `excludeFluentList.txt <https://github.com/ansys/pyfluent/blob/main/docker/fluent/excludeFluentList.txt>`_

1. We have excluded these files because we have determined that they are not needed to run typical Fluent workflows.

2. If you find that some of the excluded files are need to run your workflows then you can remove those files from the exclusion list and repeat the build process to create a new image.

> Build Docker image
> ++++++++++++++++++

Reviewer: ```suggestion
Build the Docker image
+++++++++++++++++++++
```

Author: Done. Thank you.

Execute this command to build the Docker image:

.. code:: console

    sudo docker build -t ansys_inc:v241 .

The Docker container configuration needed to build the container image is described in the
`Dockerfile <https://github.com/ansys/pyfluent/blob/main/docker/fluent/Dockerfile>`_.

Run Docker container using the command line
-------------------------------------------

> When you execute the command for running the Docker container you must specify the Ansys license file.

Reviewer: ```suggestion
When you run the Docker container, you must specify the Ansys license file.
```

Author: Done. Thank you.
